code-challenges.test.js

```diff
@@ -10,45 +10,140 @@
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
-
-
-
-// b) Create the function that makes the test pass.
-
-
-
-// --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
-// a) Create a test with expect statements for each of the variables provided.
-
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67 8, 24]
-// Expected output: [5, 109]
-
-
-
-// b) Create the function that makes the test pass.
-
-
-
-// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
-
-// a) Create a test with an expect statement using the variables provided.
-
+// --------------------1) Create a function that takes in an array, removes the first item from the array and
+// shuffles the remaining content.
+
+// a) Create a test with an expect statement using the variable provided. HINT: Check out this resource:
+//  https://jestjs.io/docs/expect#expectarraycontainingarray
+//describe block ("", () => {
+
+  describe("rmvShuffleArr", () => {
+    //test ("", () => {
+    test("takes the first item from the array and shuffles the remaining", () => {
+    //expect() })})
+    expect(shuffleArray(colors1)).toContain("yellow", "blue", "pink", "green")
+    expect(shuffleArray(colors2)).toContain("indigo", "periwinkle", "ochre", "aquamarine", "saffron")
+    })
+  })
+
+// var colors1 = ["purple", "blue", "green", "yellow", "pink"]
+// // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+// var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+// // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+//
+//
+//
+// // b) Create the function that makes the test pass.
+//
+// delcare function rmvShuffleArr
+function rmvElementShuffle(arr) {
+  // declare new var to compare
+  let currInd = arr.length,  ranInd; // I thought I could use .length to isolate the fist index but that didn't work. I also thought
+  // I could make currInd sort as well all in one method (currInd = arr.sort().length) but It kept returning undefined for whatever
+  // shifted. I'm figuring its because of my ignorance of .length.
+
+  // Identify what is left to be shuffled by using comparitives
+  while (currInd != 0) {
+
+    //use math. methods to randomize the shuffle
+    ranInd = Math.floor(Math.random() * currInd);
+    currInd--;
+
+    // swap results with current element and move to the next
+    [arr[currInd], arr[ranInd]] = [
+      arr[ranInd], arr[currInd]];
+  }
+  // return the array, shuffled less the first element.
+  return arr.shift(); // this is the only way I could get it to come out correctly.
```

Comment:
Do you want this to be the first thing that happens so it always gets the zeroth index of given array?